app/models/dor_object_workflow_status.rb

```diff
@@ -0,0 +1,24 @@
+class DorObjectWorkflowStatus
+  attr_reader :pid
+
+  ##
+  # @param [String] pid in format "druid:abc123def4567"
+  def initialize(pid)
+    @pid = pid
+  end
+
+  ##
+  # @return [Boolean]
+  def can_open_version?
+    return false unless workflow.get_lifecycle('dor', pid, 'accessioned')
+    return false if workflow.get_active_lifecycle('dor', pid, 'submitted')
+    return false if workflow.get_active_lifecycle('dor', pid, 'opened')
+    true
+  end
+
+  private
+
+  def workflow
+    Dor::WorkflowService
```

Comment:
one other suggestion:  can we switch to `Dor::Config.workflow.client` instead of `Dor::WorkflowService`?  since that appears to be the preferred pattern now.